tests/route/test_route_consistency.py

```diff
@@ -28,16 +66,34 @@ def extract_dest_ips(self, route_entries):
     def get_route_prefix_snapshot_from_asicdb(self, duthosts):
         prefix_snapshot = {}
         max_prefix_cnt = 0
+
+        def retrieve_route_snapshot(asic, prefix_snapshot, dut_instance_name, signal_queue):
+            prefix_snapshot[dut_instance_name] = \
+                set(self.extract_dest_ips(asic.run_sonic_db_cli_cmd('ASIC_DB KEYS *ROUTE_ENTRY*')['stdout_lines']))
+            logger.debug("snapshot of route table from {}: {}".format(dut_instance_name,
+                                                                      len(prefix_snapshot[dut_instance_name])))
+            signal_queue.put(1)
+
+        thread_count = 0
+        signal_queue = queue.Queue()
         for idx, dut in enumerate(duthosts.frontend_nodes):
             for asic in dut.asics:
                 dut_instance_name = dut.hostname + '-' + str(asic.asic_index)
                 if dut.facts['switch_type'] == "voq" and idx == 0:
                     dut_instance_name = dut_instance_name + "UpstreamLc"
-                prefix_snapshot[dut_instance_name] = \
-                    set(self.extract_dest_ips(asic.run_sonic_db_cli_cmd('ASIC_DB KEYS *ROUTE_ENTRY*')['stdout_lines']))
-                logger.debug("snapshot of route table from {}: {}".format(dut_instance_name,
-                                                                          len(prefix_snapshot[dut_instance_name])))
-                max_prefix_cnt = max(max_prefix_cnt, len(prefix_snapshot[dut_instance_name]))
+                    threading.Thread(target=retrieve_route_snapshot, args=(asic, prefix_snapshot,
+                                                                           dut_instance_name, signal_queue)).start()
+                    thread_count += 1
+
+        ts1 = time.time()
+        while signal_queue.qsize() < thread_count:
+            ts2 = time.time()
+            if (ts2 - ts1) > 60:
```

Comment:
just curious, how do you come up with the asic db timeout to 60 sec here?

Reply:
just randomly picked a number that I think was sufficiently large.